docs/specs/servers/blockchain/blockchain-server-api.md

```diff
@@ -0,0 +1,91 @@
+# Blockchain API Server
+
+## Blockchain RPC
+
+Blockchain RPC API for interacting with different blockchains can be found in a separate `Blockchain RPC` subsection.
+
+## Hexless account profiles API
+
+Email wallet users can register a profile name e.g. `coolname.connect.id` and use it instead of the `0x` address.
+Server API exposes forward lookup for the `name.connect.id` -> `0x` address in different chains and reverse lookup for the `0x` -> `name.connect.id`.
+Users can also update the name data including avatar, description text and bind addresses.
+
+### Name lookup
+
+Used to lookup chains addresses for the name.
+
+`GET /v1/profile/account/{name}`
+
+* `name` - is the registered name. eg. `coolname.connect.id`
+
+#### Success response body:
+
+```jsonc
+{
+    "name": string,
+    "text": (optional)string,
+    "avatar": (optional)string,
+    "registered": datetime,
+    "updated": datetime,
+    "addresses": {
+        "{chainID}": (H160)string
+    },
+}
+```
+
+* `name` - registered account name.
+* `text` - account profile self description.
+* `avatar` - URL of the avatar for the account.
+* `registered` - date and time of the name registration.
+* `updated` - date and time of the last name data update.
+* `addresses` - object with the list of registered addresses for `chainID`s.
+
+#### Response error codes:
+
+* `404 Not Found` - The requested name is not registered.
+* `406 Not Acceptable` - Wrong requested name format.
+
+### Reverse name lookup
+
+Used to lookup registered account name for the address.
+
+`GET /v1/profile/reverse/{address}`
+
+* `address` - is the address in `H160` format. eg. `0x71C7656EC7ab88b098defB751B7401B5f6d8976F`
+
+#### Success response body:
+
+```jsonc
+{
+    "name": string,
+    "text": (optional)string,
+    "avatar": (optional)string,
+    "chainID": integer
+}
+```
+
+* `name` - registered account profile name.
+* `text` - account profile self description.
+* `avatar` - URL of the avatar for the account.
+* `chainID` - chain id where the lookup address is bind.
+
+#### Response error codes:
+
+* `404 Not Found` - The requested address is not registered for any name.
+* `406 Not Acceptable` - Wrong requested address format.
```

Comment:
406 is not an appropriate response here, we should either use 400 or 422

Reply:
Agree! Changed to the 400 Bad Request.